File: Two_Sum_py/two_sum.py
#Non-ai two-sum implementation.
def two_sum(nums : list[int], target: int) -> list[int]:
    indx1 = 0
    indx2 = 1
    solution = []
    while indx1 < len(nums):
        if nums[indx1] + nums[indx2] == target:
            solution.append(indx1)
            solution.append(indx2)
            indx1 += 1
            indx2 = indx1 + 1
        else:
            if indx2 < len(nums):
                indx2 += 1
            else:
                break
    return solution

File: Two_Sum_Implementations/Two_Sum_rs/two_sum.rs
//Non-AI Rust implementation of two-sum.
fn two_sum(nums: &[i32], target : i32) {
    let solution: [i32, 2] = [0,0];
    let indx1: i32 = 0;
    let indx2: i32 = indx1 + 1;
    while indx1 < nums.len() {
        if nums[indx1] + nums[indx2] == target {
            solution[0] = indx1;
            solution[1] = indx2;
            return solution;
        }
        else {
            if indx2+1 == nums.len() {
                indx1 = indx1 + 1;
                indx2 = indx1 + 1;
            }
            else {
                indx2 = indx2 + 1;
            }
        }
    }
    return solution;
}